packages/nimble-components/src/table/tests/table-selection.spec.ts

```diff
@@ -0,0 +1,372 @@
+                expect(element.getSelectedRecordIds()).toEqual([]);
+            });
+
+            it('setting selection does not apply', () => {
+                const selection = ['1'];
+                element.setSelectedRecordIds(selection);
+                expect(element.getSelectedRecordIds()).toEqual([]);
+            });
+        });
+
+        describe('with selection mode of "single"', () => {
+            beforeEach(async () => {
+                element.selectionMode = TableRowSelectionMode.single;
+                element.idFieldName = 'stringData';
+                await waitForUpdatesAsync();
+            });
+
+            it('nothing is selected by default', () => {
+                expect(element.getSelectedRecordIds()).toEqual([]);
+            });
+
+            it('can get and set selection with one record ID', () => {
+                const selection = ['1'];
+                element.setSelectedRecordIds(selection);
+                expect(element.getSelectedRecordIds()).toEqual(selection);
+            });
+
+            it('does not apply selection when the record ID is not in the data set', () => {
+                const selection = ['not-there'];
+                element.setSelectedRecordIds(selection);
+                expect(element.getSelectedRecordIds()).toEqual([]);
+            });
+
+            it('filters out selection when some record IDs are not in the data set', () => {
+                const selection = ['not-there', '1'];
+                element.setSelectedRecordIds(selection);
+                expect(element.getSelectedRecordIds()).toEqual(['1']);
+            });
+
+            it('only applies first record ID', () => {
+                const selection = ['1', '2'];
+                element.setSelectedRecordIds(selection);
+                expect(element.getSelectedRecordIds()).toEqual(['1']);
+            });
+
+            it('can clear selection', () => {
+                element.setSelectedRecordIds(['1']);
+                element.setSelectedRecordIds([]);
+                expect(element.getSelectedRecordIds()).toEqual([]);
+            });
+
+            it('does not fire selection-change event when selection is changed by calling setSelectedRecordIds', () => {
+                const selectionChangeListener = createEventListener(
+                    element,
+                    'selection-change'
+                );
+                element.setSelectedRecordIds(['0']);
+                expect(selectionChangeListener.spy).not.toHaveBeenCalled();
+            });
+        });
+    });
+
+    describe('interactive selection', () => {
+        let selectionChangeListener: {
+            promise: Promise<void>,
+            spy: jasmine.Spy
+        };
+
+        beforeEach(() => {
+            selectionChangeListener = createEventListener(
+                element,
+                'selection-change'
+            );
+        });
+
+        describe('with selection mode of "none"', () => {
```

Comment:
It could be good to add a case that the selection change event doesn't fire when you click on a row with selection mode of "none".

Reply:
Right now, it is part of the "clicking a row does not select it" test, but I can split it out into its own test to make that more explicit.